src/Backups/BackupImpl.cpp

```diff
@@ -929,6 +935,12 @@ void BackupImpl::writeFile(const BackupFileInfo & info, BackupEntryPtr entry)
         writer->copyDataToFile(info.data_file_name, create_read_buffer, info.base_size, info.size - info.base_size);
     }
 
+    if (!deduplicate_files)
+    {
+        for (const auto & reference : info.reference_sources)
+            writer->copyFile(reference, info.data_file_name, info.size - info.base_size);
```

Comment:
I like this approach. And it can be faster too because instead of copying the same file again to another destination you just copy it inside a backup.